plugins/modules/ec2_instance.py

```diff
@@ -2117,6 +2201,18 @@ def main():
         ),
         tenancy=dict(type="str", choices=["dedicated", "default"]),
         placement_group=dict(type="str"),
+        placement=dict(
+            type="dict",
+            options=dict(
+                affinity=dict(type="str"),
+                availability_zone=dict(type="str"),
+                group_name=dict(type="str"),
+                host_id=dict(type="str"),
+                host_resource_attr=dict(type="str"),
+                partition_number=dict(type="int"),
+                tenancy=dict(type="str", choices=["dedicated", "default"]),
```

Comment:
This puts tenancy in two separate places, could this be taken from the tenancy setting above?

Reply:
It does, I didn't want to introduce a breaking change, unless it is ok?
The AWS SDK has the `tenancy` attribute inside the `placement` attributes, I wanted to be as close as possible to the AWS python SDK.

Reply:
We can deprecate the other option, use this if set, use the other if not, throw an error if both are set.

Reply:
done 👍